src/Compilers/Core/Portable/DiagnosticAnalyzer/DiagnosticQueue.cs

```diff
@@ -95,13 +98,14 @@ public override bool TryDequeue(out Diagnostic d)
         /// </summary>
         private sealed class CategorizedDiagnosticQueue : DiagnosticQueue
         {
-            private readonly object _gate = new object();
-            private Dictionary<DiagnosticAnalyzer, SimpleDiagnosticQueue> _lazyLocalSemanticDiagnostics;
-            private Dictionary<DiagnosticAnalyzer, SimpleDiagnosticQueue> _lazyLocalSyntaxDiagnostics;
-            private Dictionary<DiagnosticAnalyzer, SimpleDiagnosticQueue> _lazyNonLocalDiagnostics;
+            private readonly object _gate;
+            private Dictionary<DiagnosticAnalyzer, SimpleDiagnosticQueue>? _lazyLocalSemanticDiagnostics;
+            private Dictionary<DiagnosticAnalyzer, SimpleDiagnosticQueue>? _lazyLocalSyntaxDiagnostics;
+            private Dictionary<DiagnosticAnalyzer, SimpleDiagnosticQueue>? _lazyNonLocalDiagnostics;
 
             public CategorizedDiagnosticQueue()
             {
+                _gate = new object();
                 _lazyLocalSemanticDiagnostics = null;
                 _lazyLocalSyntaxDiagnostics = null;
                 _lazyNonLocalDiagnostics = null;
```

Comment:
💡  It would be easier to just remove all of these lines and leave the original initializer for `_gate` in place